base_layer/blockchain/src/error.rs

```diff
@@ -32,5 +35,16 @@ pub enum ChainError {
 /// The chainstate is used to present all generic chain error of the actual blockchain state
 #[derive(Debug, Error)]
 pub enum StateError {
-    Brokenchain, // place holder for real error
+    // could not create lmdb database
+    LmdbError(DatastoreError),
```

Comment:
Ideally, there should be no references to LMDB here; it's supposed to be abstracted away